### PR TITLE
mb/protectli/vault_adl_n: Initial support for VP32XX series

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

**Comment by miczyg1** [comment]
@SergiiDmytruk something is wrong with the defconfigs checks. It complains on the VP66XX defconfig, that 
```
CONFIG_PCIEXP_L1_SUB_STATE=y
CONFIG_PCIEXP_CLK_PM=y
```

Should be removed. But in the `src/mainboard/protectli/vault_adl_p/Kconfig` has:

```
config PCIEXP_L1_SUB_STATE
	default n

config PCIEXP_CLK_PM
	default n
```

So effectively, the change will disable these features, which is probably not desired...

**Comment by SergiiDmytruk** [comment]
Looks fine to me, because `src/soc/intel/common/block/pcie/Kconfig` has:
```
config PCIEXP_CLK_PM
	default y

config PCIEXP_L1_SUB_STATE
	default y
```
as shown in `menuconfig`:
```
...
Defined at src/mainboard/protectli/vault_adl_p/Kconfig:54
  Depends on: VENDOR_PROTECTLI [=y] && BOARD_PROTECTLI_VP66XX [=n]
...
Defined at src/soc/intel/common/block/pcie/Kconfig:16
  Depends on: SOC_INTEL_COMMON [=y] && SOC_INTEL_COMMON_BLOCK [=y] && SOC_INTEL_COMMON_BLOCK_PCIE [=y]
...
```

**Comment by miczyg1** [comment]
> Looks fine to me, because `src/soc/intel/common/block/pcie/Kconfig` has:
> 
> ```
> config PCIEXP_CLK_PM
> 	default y
> 
> config PCIEXP_L1_SUB_STATE
> 	default y
> ```
> 
> as shown in `menuconfig`:
> 
> ```
> ...
> Defined at src/mainboard/protectli/vault_adl_p/Kconfig:54
>   Depends on: VENDOR_PROTECTLI [=y] && BOARD_PROTECTLI_VP66XX [=n]
> ...
> Defined at src/soc/intel/common/block/pcie/Kconfig:16
>   Depends on: SOC_INTEL_COMMON [=y] && SOC_INTEL_COMMON_BLOCK [=y] && SOC_INTEL_COMMON_BLOCK_PCIE [=y]
> ...
> ```

It does not look fine to me. Say, I remove those lines from the Kconfig. What will happen?

1. Default values from `src/soc/intel/common/block/pcie/Kconfig`  take priority? OR
2. Default values from `src/mainboard/protectli/vault_adl_p/Kconfig` take priority?

I thought the defaults in mainboard's Kconfig will override the defaults from the rest of Kconfig. But maybe I am mistaken.

**Comment by SergiiDmytruk** [comment]
`src/mainboard/protectli/vault_adl_p/Kconfig` doesn't apply because `BOARD_PROTECTLI_VP66XX` isn't set.

**Comment by miczyg1** [comment]
> There is no order dependence, `src/mainboard/protectli/vault_adl_p/Kconfig` doesn't apply because `BOARD_PROTECTLI_VP66XX` isn't set.

:facepalm: alright... Tha make `olddefconfig` didn't select it previously because vault_adl_p was first in the protectli dir, now it isn't...

**Review by krystian-hebel — CHANGES_REQUESTED** [review]

File: src/mainboard/protectli/vault_adl_n/gpio.c [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+
+	/* GPP_H0 - GPIO */
+	PAD_CFG_GPO(GPP_H0, 0, DEEP),
+	/* GPP_H1 - GPIO */
+	PAD_CFG_GPO(GPP_H1, 0, DEEP),
+	/* GPP_H2 - GPIO */
+	PAD_CFG_GPO(GPP_H2, 0, DEEP),
+	PAD_NC(GPP_H3, NONE),
+	PAD_NC(GPP_H4, NONE),
+	PAD_NC(GPP_H5, NONE),
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_adl_n/gpio.c:167]: NC. What else?

File: src/mainboard/protectli/vault_adl_n/Makefile.inc [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+bootblock-y += bootblock.c
+
+romstage-y += gpio.c
+romstage-y += romstage_fsp_params.c
+
+ramstage-y += mainboard.c
+
+bootblock-y += die.c
+romstage-y += die.c
```
Comment (mkopec) [inline src/mainboard/protectli/vault_adl_n/Makefile.inc:11]: might also want to add it to ramstage and smm like the other protectli platforms other than adl-p

Comment (mkopec) [inline src/mainboard/protectli/vault_adl_n/Makefile.inc:11]:
```suggestion
smm-y += die.c
smm-y += ec.
```

Comment (miczyg1) [reply src/mainboard/protectli/vault_adl_n/Makefile.inc:11]: Not sure if we want to "die" with beeps in SMM.

File: src/mainboard/protectli/vault_adl_n/board_info.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Category: sbc
+Board URL: TBD
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_adl_n/board_info.txt:2]: Actually the VP2430 will also be a variant here, so not sure how to put all links

[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_adl_n/board_info.txt]: 5a6373ffddc9f7be230179dc8bd5aeed334f9570

File: src/mainboard/protectli/vault_adl_n/Makefile.inc (hunk 2 above)
Comment (miczyg1) [reply src/mainboard/protectli/vault_adl_n/Makefile.inc]: 174bb97e2aa48b566a2610c73c1b4a201d512d53

File: src/mainboard/protectli/vault_adl_n/gpio.c [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+	PAD_CFG_NF(GPP_B12, NONE, DEEP, NF1),
+	/* GPP_B13 - PLTRST# */
+	PAD_CFG_NF(GPP_B13, NONE, DEEP, NF1),
+	/* GPP_B14 - SPKR */
+	PAD_CFG_NF(GPP_B14, NONE, PLTRST, NF1),
+	PAD_NC(GPP_B15, NONE),
+	/* GPP_B16 - I2C5_SDA */
+	PAD_CFG_NF(GPP_B16, NONE, DEEP, NF2),
+	/* GPP_B17 - I2C5_SCL */
+	PAD_CFG_NF(GPP_B17, NONE, DEEP, NF2),
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_adl_n/gpio.c]: 89bad108d746d4a42e11ab7bb7479a7d8feeb2c3

File: src/mainboard/protectli/vault_adl_n/Makefile.inc (hunk 2 above)
Comment (miczyg1) [reply src/mainboard/protectli/vault_adl_n/Makefile.inc]: Maybe... Added to all stages, including SMM, anyways.

**Review by mkopec — DISMISSED** [review]
LGTM, some comments @krystian-hebel remain though

File: src/mainboard/protectli/vault_adl_n/bootblock.c [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	pnp_exit_conf_state(GPIO_DEV);
+}
+
+static void ite_gpio_setup(u8 gpio, u8 polarity, u8 pullup, u8 output, u8 enable)
+{
+	u8 set = (gpio / 10) - 1;
+	u8 pin = (gpio % 10);
+
+	/* There are only 6 configurable sets, 8 pins each */
+	if (gpio < 10 || set > 6 || pin > 7)
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply src/mainboard/protectli/vault_adl_n/bootblock.c]: Then why is there `-1` in the first line of this function? IIUC GPIO numbers start with 10, which would result in `set` equal to 0.

[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_adl_n/bootblock.c]: Ahh right... I missed it.

**Review by krystian-hebel — DISMISSED** [review]

**Review by SergiiDmytruk — APPROVED** [review]

**Pull request merged**